pkg/dns/treecache.go

```diff
@@ -199,10 +143,6 @@ func (cache *TreeCache) deletePath(path ...string) bool {
 }
 
 func (cache *TreeCache) deleteEntry(key string, path ...string) bool {
-	if cache.rwLock != nil {
```

Comment:
The comments document the assumptions.  A disconnected tree doesn't need
locks, but it's WAY too easy to forget what the locking protocol is without
comments.

On Fri, May 20, 2016 at 10:49 PM, Abhi Shah <notifications@github.com>
wrote:

> In pkg/dns/treecache.go
> <https://github.com/kubernetes/kubernetes/pull/23930#discussion_r64129291>
> :
>
> > @@ -199,10 +143,6 @@ func (cache *TreeCache) deletePath(path ...string) bool {
> >  }
> >
> >  func (cache *TreeCache) deleteEntry(key string, path ...string) bool {
> > -	if cache.rwLock != nil {
>
> I don't think the comment would make sense here. the treecache is not
> thread-safe.
> It could be used by the same thread/go-routine without locking,
> or by multiple threads, using a simple mutex or rwmutex.
> it is up to the caller to choose..
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23930/files/0f248f9b1173265f2fd3524f193e685ae712eb89..2422e1cea35c140337e64708f33d0cf8d5f053fc#r64129291>
>